tests/json_tests.rs

```diff
@@ -53,57 +54,57 @@ fn test_flowtable() {
     //  add rule inet some_inet_table forward ct state established flow add @flowed'
     // ```
     let expected: Nftables = Nftables {
-        objects: vec![
-            NfObject::ListObject(Box::new(NfListObject::Table(Table {
+        objects: Cow::Borrowed(&[
+            NfObject::ListObject(NfListObject::Table(Table {
                 family: NfFamily::INet,
-                name: "some_inet_table".to_string(),
+                name: Cow::Borrowed("some_inet_table"),
                 handle: None,
-            }))),
-            NfObject::ListObject(Box::new(NfListObject::FlowTable(FlowTable {
+            })),
+            NfObject::ListObject(NfListObject::FlowTable(FlowTable {
                 family: NfFamily::INet,
-                table: "some_inet_table".to_string(),
-                name: "flowed".to_string(),
+                table: Cow::Borrowed("some_inet_table"),
+                name: Cow::Borrowed("flowed"),
                 handle: None,
                 hook: Some(NfHook::Ingress),
                 prio: Some(0),
-                dev: Some(vec!["lo".to_string()]),
-            }))),
-            NfObject::ListObject(Box::new(NfListObject::Chain(Chain {
+                dev: Some(Cow::Borrowed(&[Cow::Borrowed("lo")])),
+            })),
+            NfObject::ListObject(NfListObject::Chain(Chain {
                 family: NfFamily::INet,
-                table: "some_inet_table".to_string(),
-                name: "some_inet_chain".to_string(),
+                table: Cow::Borrowed("some_inet_table"),
+                name: Cow::Borrowed("some_inet_chain"),
                 newname: None,
                 handle: None,
                 _type: Some(NfChainType::Filter),
                 hook: Some(NfHook::Forward),
                 prio: None,
                 dev: None,
                 policy: Some(NfChainPolicy::Accept),
-            }))),
-            NfObject::ListObject(Box::new(NfListObject::Rule(Rule {
+            })),
+            NfObject::ListObject(NfListObject::Rule(Rule {
                 family: NfFamily::INet,
-                table: "some_inet_table".to_string(),
-                chain: "some_inet_chain".to_string(),
-                expr: vec![
+                table: Cow::Borrowed("some_inet_table"),
+                chain: Cow::Borrowed("some_inet_chain"),
```

Comment:
Would it also be possible to use `.into()` in places like these?

Reply:
`impl<'a> From<&'a str> for Cow<'a, str>` (and its fellows) is not `const` for now so `.into()` will result in a temporary value (as seen by rustc), so the resulting container is not `'static`, therefore cannot be borrowed `'static`. In this example both the outer slice and the inner `str` needs to be borrowed `'static`, leading to the following error:

```
error[E0716]: temporary value dropped while borrowed
   --> tests/json_tests.rs:57:33
    |
57  |            objects: Cow::Borrowed(&[
    |   ________________________________-^
    |  |_________________________________|
58  | ||             NfObject::ListObject(NfListObject::Table(Table {
59  | ||                 family: NfFamily::INet,
60  | ||                 name: Cow::Borrowed("some_inet_table"),
...   ||
106 | ||             })),
107 | ||         ]),
    | ||         ^
    | ||_________|
    |  |_________creates a temporary value which is freed while still in use
    |            cast requires that borrow lasts for `'static`
108 |        };
    |         - temporary value is freed at the end of this statement
```

If all of the outer containers are `Cow::Owned`, we can use `.into()` inside. Otherwise we probably need to stick to manually use `Cow::Borrowed`. I've also pushed a commit that allows temporary arrays to be used by replacing `'static` with `'a`. It will then just need to move the array in `Cow::Borrowed(&[..])` to another local variable to allow the usage of `.into()`:

```rust
let objects = [
    NfObject::ListObject(NfListObject::Table(Table {
        family: NfFamily::INet,
        name: "some_inet_table".into(), // here
        handle: None,
    })),
    // ...
];
let expected = Nftables {
    objects: objects.into(); // and here
};
```